lib/ui/window/platform_message_response_dart.cc

```diff
@@ -63,16 +73,44 @@ PlatformMessageResponseDart::~PlatformMessageResponseDart() {
 }
 
 void PlatformMessageResponseDart::Complete(std::unique_ptr<fml::Mapping> data) {
-  PostCompletion(std::move(callback_), ui_task_runner_, &is_complete_, channel_,
-                 [data = std::move(data)] {
-                   return tonic::DartByteData::Create(data->GetMapping(),
-                                                      data->GetSize());
-                 });
+  if (send_port_ == ILLEGAL_PORT) {
+    PostCompletion(std::move(callback_), ui_task_runner_, &is_complete_,
+                   channel_, [data = std::move(data)] {
+                     return tonic::DartByteData::Create(data->GetMapping(),
+                                                        data->GetSize());
+                   });
+  } else {
+    // TODO(gaaclarke): Move all this logic into its own
+    // PlatformMessageResponse.
+    is_complete_ = true;
+    Dart_CObject response = {
+        .type = Dart_CObject_kExternalTypedData,
+    };
+    uint8_t* copy = static_cast<uint8_t*>(malloc(data->GetSize() + 8));
```

Comment:
Not sure how big those bytes get here, but you can avoid the copy if you want.

Since `data` is actually `std::unique_ptr` it means we are the only owner of the data (and would discard it here when going out of scope). So you can
```
void FreeFinalizer(void* isolate_callback_data, void* peer) {
  delete reinterpret_cast<fml::Mapping*>(peer);
}

fml::Mapping* mapping = data.release();
Dart_CObject bytes;
bytes.value.as_external_typed_data.type = Dart_TypedData_kUint8;
bytes.value.as_external_typed_data.data = mapping->GetMapping();
bytes.value.as_external_typed_data.length = mapping->GetSize();
bytes.value.as_external_typed_data.peer = mapping; // VM takes ownership of "mapping" and will call finalizer.
bytes.value.as_external_typed_data.callback = FreeFinalizer;
```

Reply:
I have an outstanding PR where I remove this copy for the old code.  It's actually quite involved since Mapping can point to data you can't verify is mutable or single-owner.  Or it is memmap'd in a way that will crash if you write to it.  I solved that on some platforms by memmaping with copy-on-write, which isn't implemented yet on fuchsia.  Anyways it's been a headache getting it approved so I'm going to avoid that for now.